File: hello-go/src/logger/logger.go
package logger

import (
	"fmt"
	"time"
)

var verbose bool

func Init(v bool) {
	verbose = v
}

func formatTimestamp() string {
	return time.Now().Format("2006-01-02 15:04:05.000")
}

func Debug(message string) {
	if verbose {
		fmt.Printf("[%s] [debug] %s\n", formatTimestamp(), message)
	}
}

func Info(message string) {
	fmt.Printf("[%s] [info] %s\n", formatTimestamp(), message)
}

func Warn(message string) {
	fmt.Printf("[%s] [warn] %s\n", formatTimestamp(), message)
}

func Error(message string) {
	fmt.Printf("[%s] [error] %s\n", formatTimestamp(), message)
}

func Phase(phase string) {
	fmt.Println()
	fmt.Printf("[%s] [info] === %s ===\n", formatTimestamp(), phase)
}

File: hello-python/src/audio_client/logger.py
"""Logging utility with support for different log levels"""

from datetime import datetime

_verbose_enabled = False


def init(verbose: bool):
    """Initialize logger with verbose flag"""
    global _verbose_enabled
    _verbose_enabled = verbose


def _format_timestamp() -> str:
    """Format current timestamp"""
    return datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")[:-3]


def info(message: str):
    """Log info message"""
    print(f"[{_format_timestamp()}] [info] {message}")


def error(message: str):
    """Log error message"""
    print(f"[{_format_timestamp()}] [error] {message}")


def warn(message: str):
    """Log warning message"""
    print(f"[{_format_timestamp()}] [warn] {message}")


def warning(message: str):
    """Log warning message (alias for warn)"""
    warn(message)


def debug(message: str):
    """Log debug message (only if verbose enabled)"""
    if _verbose_enabled:
        print(f"[{_format_timestamp()}] [debug] {message}")


def phase(message: str):
    """Log phase separator"""
    print()
    print(f"[{_format_timestamp()}] [info] === {message} ===")
